src/effects/backends/builtin/pitchshifteffect.cpp

```diff
@@ -61,12 +71,69 @@ EffectManifestPointer PitchShiftEffect::getManifest() {
     pitch->setNeutralPointOnScale(0.0);
     pitch->setRange(-1.0, 0.0, 1.0);
 
+    EffectManifestParameterPointer rangeMode = pManifest->addParameter();
+    rangeMode->setId("rangeMode");
+    rangeMode->setName(QObject::tr("Mode"));
+    rangeMode->setShortName(QObject::tr("Mode"));
+    rangeMode->setDescription(QObject::tr(
```

Comment:
IMO this is knob is unnecessary. What was your motivation for introducing it?

Reply:
The idea is based on the demo of the Pioneer DJM-900NX2 build-in effects including the Pitch Shift effect. As is shown in the [demo video](https://youtu.be/9BTKCT7NHIM?t=964), with turning the "TIME" knob, using the effect "LEVEL/DEPTH" knob, the Pitch goes only up or down based on the mentioned "TIME" knob setting (IMO it is not to best understandable solution, but don't care). Based on that idea, I think that it can be useful to work with the pitch in only one way (up or down) and that working with the knob in his limit bounds (minimum bound and maximum bound) is much easier. You don't have to take care to go from the middle knob position to maximum and back (and not go under the middle position). With that, the Pitch knob is much more sensitive, because the range, which was previously only the half of range (from the middle position to maximum for pitching up) is now recalculated to the whole knob range. In summary, it looks cool to me to allow working with the effect as same as on the Pioneer mixer (and maybe some other brand's mixers too), but that is just my point of view.

I agree, that using the knob for only 3 possible settings is not the clearest solution, but it cannot be allowed more modes simultaneously.

Reply:
The DJM-900NX2 TIME knob behaves differently. It is our original Pitch knob with unity at center. It sets the range of the effect when using the (additional) "LEVEL/DEPTH" knob.
So instead of introducing "Mode" and "wideRange" you may just introduce a "Range" knob, that set the min/max of the Pitch knob. 

This maps to the Pioneer knobs like this: 
* TIME = Pitch 
* "LEVEL/DEPTH" = Range

Reply:
Yes, but the range knob has no regions, it is continuously. That's why Phil Harris eagerly turns it to be sure to hit the max range. The manual says it has a Rage from -50 ... +100%  whatever that means. Translated into speed it would be -12 ... +12 semitones. https://docs.pioneerdj.com/Manuals/DJM_900NXS2_DRI1300A_manual/?_ga=2.239935050.831276606.1638182063-764262911.1599494699&page=16
Does that make sense? I think it is much more when listening to the video. 
The proposed -24 to + 24 sound reasonable.

Reply:
Perfect, thank you so much. I think I understand it completely now.